### DISCUSSION TITLE
How to add custom fields to bibliography entries?

### DISCUSSION THREAD

**Discussion by kirk86** [alshedivat/al-folio#642] [Q&A]
Hi there, thanks for the great template!

I have a question to which I couldn't find the answer anywhere on the project readme or faq.

From your website seems that is possible to have multiple abbreviations, e.g.
<img width="100" alt="image" src="https://user-images.githubusercontent.com/2902390/161449906-1ba035f8-d4b3-4217-96c4-35eaeb45582b.png">

1. What's the custom field that should be added in the bib file to get this result?

2. What is the custom field to add the `talk` or `video` under a reference, like for instance

<img width="447" alt="image" src="https://user-images.githubusercontent.com/2902390/161449984-ffcfc3a6-d023-464e-a23c-23bcb8a63d05.png">

I've tried using both `oral` and `talk` but to no avail.

Maybe all the possible custom fields should be in one place somewhere on the readme easily accessible by everyone?

Thanks.

**Comment by Luosuu** [comment]
I have same question here.

**Comment by alshedivat** [comment]
(converted to discussion. please use discussions for general questions. issues are mainly for reporting bugs and feature requests.)

to add a new field to bibliography items similar to the ones already there, you would need to take a couple of steps:

1. come up with a name for your field (eg, let's call it `abbr2`)
2. add it the bib entries in your bibliography file (eg, `abbr2={<something>}`)
3. now, you will have to add this new field to [`_layouts/bib.html`](https://github.com/alshedivat/al-folio/blob/master/_layouts/bib.html):
  - figure out which field you want to make it similar to. if you want it to be similar to `abbr`, find the part of the code in `_layouts/bib.html` that defines `abbr`
  - duplicate the code (simplify, if necessary) and change `abbr` to `abbr2`. eg, add something like this right after the `abbr` part:
```
{%- if entry.abbr2 -%}
  <span class="abbr2 badge">{{entry.abbr2}}</span>
{%- endif -%}
```
  - if you want to, add a new style for `abbr2` class to [`_sass/_base.scss`](https://github.com/alshedivat/al-folio/blob/master/_sass/_base.scss)

to add new buttons similar to `code`, `talk`, etc. you can follow the same steps as above, just make sure to copy and edit the code of the corresponding elements. hope this helps.
